hydra-node/src/Hydra/API/HTTPServer.hs

```diff
@@ -338,6 +372,69 @@ handleSideLoadSnapshot putClientInput body = do
       putClientInput $ SideLoadSnapshot snapshot
       pure $ responseLBS status200 jsonContent (Aeson.encode $ Aeson.String "OK")
 
+-- | Handle request to submit a hydra transaction to the head.
+handleSubmitHydraTx ::
+  forall tx.
+  IsChainState tx =>
+  (ClientInput tx -> IO ()) ->
+  NominalDiffTime ->
+  TChan (Either (TimedServerOutput tx) (ClientMessage tx)) ->
+  LBS.ByteString ->
+  IO Response
+handleSubmitHydraTx putClientInput apiTransactionTimeout responseChannel body = do
+  case Aeson.eitherDecode' @(SubmitHydraTxRequest tx) body of
+    Left err ->
+      pure $ responseLBS status400 jsonContent (Aeson.encode $ Aeson.String $ pack err)
+    Right SubmitHydraTxRequest{submitHydraTx} -> do
+      -- Submit the transaction to the head
+      putClientInput (NewTx submitHydraTx)
+
+      let txid = txId submitHydraTx
+      result <- timeout (realToFrac apiTransactionTimeout) (waitForTransactionResult txid)
+
+      case result of
+        Just (SubmitHydraTxConfirmed snapshotNumber) ->
+          pure $ responseLBS status200 jsonContent (Aeson.encode $ SubmitHydraTxConfirmed snapshotNumber)
+        Just (SubmitHydraTxInvalidResponse validationError) ->
+          pure $ responseLBS status400 jsonContent (Aeson.encode $ SubmitHydraTxInvalidResponse validationError)
+        Just SubmitHydraTxSubmitted ->
+          pure $ responseLBS status202 jsonContent (Aeson.encode SubmitHydraTxSubmitted)
+        Nothing ->
+          -- Timeout occurred - return 202 Accepted with timeout info
+          pure $
+            responseLBS
+              status202
+              jsonContent
+              ( Aeson.encode $
+                  object
+                    [ "tag" .= Aeson.String "SubmitHydraTxSubmitted"
+                    , "timeout" .= Aeson.String ("Transaction submission timed out after " <> pack (show apiTransactionTimeout) <> " seconds")
+                    ]
+              )
+ where
+  --  Wait for transaction result by listening to events
+  waitForTransactionResult :: TxIdType tx -> IO SubmitHydraTxResponse
```

Comment:
I think this is a great idea and we should leverage in other endpoints too!

There is even an [open issue](https://github.com/cardano-scaling/hydra/issues/1911) about this.

Nice job 👏